megamek/src/megamek/client/ui/swing/boardview/StepSprite.java

```diff
@@ -89,41 +89,31 @@ public void prepare() {
         boolean isLastLegalStep = isLastStep &&
                 (step.getMovementType(true) != EntityMovementType.MOVE_ILLEGAL);
 
-        Color col;
         boolean jumped = false;
-        // Choose the color according to the type of movement
-        switch (step.getMovementType(isLastStep)) {
+        boolean isMASCOrSuperCharger = false;
+        boolean isBackwards = false;
+
+        EntityMovementType movementType = step.getMovementType(isLastStep);
+        switch (movementType) {
             case MOVE_RUN:
             case MOVE_VTOL_RUN:
             case MOVE_OVER_THRUST:
-                if (step.isUsingMASC() || step.isUsingSupercharger()) {
-                    col = GUIP.getColor("AdvancedMoveMASCColor");
-                } else {
-                    col = GUIP.getColor("AdvancedMoveRunColor");
-                }
+                isMASCOrSuperCharger = (step.isUsingMASC() || step.isUsingSupercharger());
                 break;
             case MOVE_JUMP:
                 jumped = true;
-                col = GUIP.getColor("AdvancedMoveJumpColor");
-                break;
-            case MOVE_SPRINT:
-            case MOVE_VTOL_SPRINT:
-                col = GUIP.getColor("AdvancedMoveSprintColor");
-                break;
-            case MOVE_ILLEGAL:
-                col = GUIP.getColor("AdvancedMoveIllegalColor");
                 break;
             default:
                 if ((step.getType() == MoveStepType.BACKWARDS)
                         || (step.getType() == MoveStepType.LATERAL_LEFT_BACKWARDS)
                         || (step.getType() == MoveStepType.LATERAL_RIGHT_BACKWARDS)) {
-                    col = GUIP.getColor("AdvancedMoveBackColor");
-                } else {
-                    col = GUIP.getColor("AdvancedMoveDefaultColor");
+                    isBackwards = true;
                 }
                 break;
         }
 
+        Color col = GUIP.getColorForMovement(step.getMovementType(isLastStep), isMASCOrSuperCharger, isBackwards);
```

Comment:
```suggestion
        Color col = GUIP.getColorForMovement(movementType, isMASCOrSuperCharger, isBackwards);
```

I believe this is the original intention